src/backend/base/langflow/helpers/flow.py

```diff
@@ -257,23 +258,23 @@ def get_arg_names(inputs: List["Vertex"]) -> List[dict[str, str]]:
     ]
 
 
-def get_flow_by_id_or_endpoint_name(
-    flow_id_or_name: str, db: Session = Depends(get_session), user_id: Optional[UUID] = None
-) -> Flow:
-    endpoint_name = None
-    try:
-        flow_id = UUID(flow_id_or_name)
-        flow = db.get(Flow, flow_id)
-    except ValueError:
-        endpoint_name = flow_id_or_name
-        stmt = select(Flow).where(Flow.endpoint_name == endpoint_name)
-        if user_id:
-            stmt = stmt.where(Flow.user_id == user_id)
-        flow = db.exec(stmt).first()
-    if flow is None:
-        raise HTTPException(status_code=404, detail=f"Flow identifier {flow_id_or_name} not found")
-
-    return flow
+def get_flow_by_id_or_endpoint_name(flow_id_or_name: str, user_id: Optional[UUID] = None) -> FlowRead | None:
```

Comment:
@ogabrielluiz could you explain why `Depends(get_session)` wouldn't release the session?